text/0000-default-field-values.md

```diff
@@ -0,0 +1,1658 @@
+    speed: f64,
+    ...
+}
+```
+
+By having default field values in the language, `structopt` could let you write:
+
+```rust
+#[derive(Debug, StructOpt)]
+#[structopt(name = "example", about = "An example of StructOpt usage.")]
+struct Opt {
+    /// Set speed
+    #[structopt(short = "s", long = "speed")]
+    speed: f64 = 42,
+    ...
+}
+```
+
+### `derive_builder`
+
+[`derive_builder`]: https://docs.rs/derive_builder/0.7.0/derive_builder/#default-values
+
+A third example comes from the crate [`derive_builder`]. As the name implies,
+you can use it to `#[derive(Builder)]`s for your types. An example is:
+
+```rust
+#[derive(Builder, Debug, PartialEq)]
+struct Lorem {
+    #[builder(default = "42")]
+    pub ipsum: u32,
+}
+```
+
+### `proptest_derive`
+
+The yet to be released `#[derive(Arbitrary)]` custom derive macro also has a
+notion of defaults; for example, you may write:
+
+```rust
+#[derive(Debug, Arbitrary)]
+struct Foo {
+    bar: usize,
+    #[proptest(value = "42")]
+    baz: usize,
+}
+```
+
+This will generate random `usize`s for `bar` and fix the value of `baz` to `42`.
+One could instead imagine writing:
+
+```rust
+#[derive(Debug, Arbitrary)]
+struct Foo {
+    bar: usize,
+    #[proptest(default)]
+    baz: usize = 42,
+}
+```
+
+This would then fix the value of `baz` to `42`.
+
+### Conclusion
+
+As seen in the previous sections, rather than make deriving `Default`
+more magical, by allowing default field values in the language,
+user-space custom derive macros can make use of them.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+## Providing field defaults
+
+Consider a data-type such as (1):
+
+```rust
+pub struct Probability {
+    value: f32,
+}
+```
+
+You'd like encode the default probability value to be `0.5`;
+With this RFC now you can provide such a default directly where `Probability`
+is defined like so (2):
+
+```rust
+pub struct Probability {
+    value: f32 = 0.5,
+}
+```
+
+Having done this, you can now construct a `Probability` with a struct
+initializer and leave `value` out to use the default (3):
+
+```rust
+let prob = Probability { .. };
+```
+
+## Deriving `Default`
+
+Previously, you might have instead implemented the `Default` trait like so (4):
+
+```rust
+impl Default for Probability {
+    fn default() -> Self {
+        Self { value: 0.5 }
+    }
+}
+```
+
+You can now shorten this to (5):
+
+```rust
+impl Default for Probability {
+    fn default() -> Self {
+        Self { .. }
+    }
+}
+```
+
+However, since you had specified `value: f32 = 0.5` in the definition of
+`Probability`, you can take advantage of that to write the more simpler
+and more idiomatic (6):
+
+```rust
+#[derive(Default)]
+pub struct Probability {
+    value: f32 = 0.5,
+}
+```
+
+Having done this, a `Default` implementation equivalent to the one in (5)
+will be generated for you.
+
+## More fields
+
+As you saw in the [summary], you are not limited to a single field and all
```

Comment:
`field, and` -- how could you forget an Oxford comma?!

Reply:
Actually, I see you've not used it multiple times, though you've also used it several times. I'll let you decide how to make it consistent.

Reply:
An Oxford comma is only necessary when you have >= 3 items in the enumeration, but if you have 2 items then no comma is needed.